report/renderer/report-utils.js

```diff
@@ -187,6 +190,77 @@ class ReportUtils {
     return clone;
   }
 
+  /**
+   * Given an audit's details, identify and return a URL locator function that
+   * can be called later with an `item` to extract the URL of it.
+   * @param {LH.FormattedIcu<LH.Audit.Details.TableColumnHeading[]>} headings
+   * @return {{(item: LH.FormattedIcu<LH.Audit.Details.TableItem>): string|undefined}=}
+   */
+  static getUrlLocatorFn(headings) {
+    // The most common type, valueType=url.
+    const urlKey = headings.find(heading => heading.valueType === 'url')?.key;
+    if (urlKey) {
+      // Return a function that extracts item.url.
+      return (item) => item[urlKey]?.toString();
```

Comment:
Shouldn't `item[urlKey]` always be a string? Do we need this `toString()`?

Reply:
I think this is better written as in L213 below.